docs/computing/mpip.md

```diff
@@ -0,0 +1,198 @@
+# mpiP: Lightweight MPI Profiler
+
+[mpiP](http://mpip.sourceforge.net/) is a lightweight MPI profiler which uses statistical sampling to record profiling data. At the momenmt is only available on Mahti. It generates less overhead and much less data than tracing tools. In order to use no code changes are require, only a re-link is needed. For both Mahti and Puhti, in addition to the required modules, one has the load the mpiP module: 
+```
+module load mpip
+```
+Next the code is compiled as normal, but with addition of the `-g` flag and the link to the libraries. For Mahti one should use:
+```
+-lmpiP -lm -lbfd -liberty -L<path-to-unwind-lib>-lunwind
+```
+The path to the unwind library on mahti is `/appl/spack/v014/install-tree/gcc-9.3.0/libunwind-1.3.1-otflii/lib`. 
+Similarly for Puhti the relink is done with :
+```
+-lmpiP -lm -L<path-to-iberty-lib>  -liberty -L<path-to-unwind-lib> -lunwind
+```
+The path to the iberty library is `/appl/spack/install-tree/intel-19.0.4/libiberty-2.31.1-o4es74/lib/`, while the path to unwind library is `/appl/spack/install-tree/intel-19.0.4/libunwind-1.2.1-45uplb/lib/`. 
+The above re-link will only work if it appears the last in the compiling line. 
+Next the code is ran as a usual batch job. The follwing additions are needed to the job script:
+```
+module load mpip
+export LD_LIBRARY_PATH=<path-to-unwind-lib>:<path-to-iberty-lib>:$LD_LIBRARY_PATH
+```
+For running instead we replace `srun` by `srun -ppdebug`. 
```

Comment:
`-ppdebug` is not a valid slurm flag,  as pdebug is not a valid partition. So that should probably be removed / replace with --partition=test or similar.

Reply:
> I tried without -ppdebug and it works, not sure why to need -ppdebug, is there a reason?

I think I took that line form the manual.